internal/job/checkout.go

```diff
@@ -726,6 +707,41 @@ func (e *Executor) defaultCheckoutPhase(ctx context.Context) error {
 	return nil
 }
 
+func gitFetchCommitWithFallback(ctx context.Context, shell *shell.Shell, gitFetchFlags, commit string) error {
+	if err := gitFetch(ctx, shell, gitFetchFlags, "origin", commit); err == nil {
+		return nil // it worked
+	} else if gerr := new(gitError); errors.As(err, &gerr) {
```

Comment:
I've wanted to clean up this particular nit for a while. Because the previous case returns:
```suggestion
	}
	if gerr := new(gitError); errors.As(err, &gerr) {
```